R/summarise_scores.R

```diff
@@ -50,7 +52,7 @@
 #' @keywords scoring
 
 summarise_scores <- function(scores,
-                             by = "model",
+                             by = character(0),
```

Comment:
I quite like the idea of keeping "model" and suggest it in the docs as a soft requirement. That way we still get a lot of nice features but there is also a lot of flexibility. Combined with a useful message if not present this seems to not have many downsides?

Reply:
Something like

```r
if (missing(by) && "model" %in% colnames(score)) {
   by <- "model"
   ## add informative message that can be suppressed by providing `by = "model"`
}
```

The downside is that the user will have to engage with this specific exception and it'll have to be documented vs. the simpler (not backwards-compatible) version of having no aggregation by default. I'd vote for doing this as a soft deprecation but phasing it out in the future.

Reply:
I was thinking we would set by to "model" as we do now but have a check that all of by are actually columns in the input

Reply:
if we do as you suggest we can detect it hasn't been set and throw a depreciation warning which you can't with my suggestion

Reply:
I would (just as Sam, I think), opt for setting `by = "model"` and then checking whether "model" is actually there.